src/core/trtserver.cc

```diff
@@ -1017,13 +1048,91 @@ TRTSERVER_ServerUnloadModel(TRTSERVER_Server* server, const char* model_name)
   ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
 
   ni::ServerStatTimerScoped timer(
-      lserver->StatusManager(), ni::ServerStatTimerScoped::Kind::CONTROL);
+      lserver->StatusManager(), ni::ServerStatTimerScoped::Kind::MODEL_CONTROL);
 
   RETURN_IF_STATUS_ERROR(lserver->UnloadModel(std::string(model_name)));
 
   return nullptr;  // success
 }
 
+TRTSERVER_Error*
+TRTSERVER_ServerRegisterSharedMemory(
+    TRTSERVER_Server* server, const char* name, const char* shm_key,
+    const size_t offset, const size_t byte_size)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(),
+      ni::ServerStatTimerScoped::Kind::SHARED_MEMORY_CONTROL);
+
+  RETURN_IF_STATUS_ERROR(lserver->RegisterSharedMemory(
+      std::string(name), std::string(shm_key), offset, byte_size));
+
+  return nullptr;  // success
+}
+
+TRTSERVER_Error*
+TRTSERVER_ServerUnregisterSharedMemory(
+    TRTSERVER_Server* server, const char* name)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(),
+      ni::ServerStatTimerScoped::Kind::SHARED_MEMORY_CONTROL);
+
+  RETURN_IF_STATUS_ERROR(lserver->UnregisterSharedMemory(std::string(name)));
+
+  return nullptr;  // success
+}
+
+TRTSERVER_Error*
+TRTSERVER_ServerUnregisterAllSharedMemory(TRTSERVER_Server* server)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(),
+      ni::ServerStatTimerScoped::Kind::SHARED_MEMORY_CONTROL);
+
+  RETURN_IF_STATUS_ERROR(lserver->UnregisterAllSharedMemory());
+
+  return nullptr;  // success
+}
+
+TRTSERVER_Error*
+TRTSERVER_ServerGetSharedMemoryStatus(
+    TRTSERVER_Server* server, std::vector<std::string>& active_shm_regions)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(),
+      ni::ServerStatTimerScoped::Kind::SHARED_MEMORY_CONTROL);
+
+  RETURN_IF_STATUS_ERROR(lserver->GetSharedMemoryStatus(active_shm_regions));
+
+  return nullptr;  // success
+}
+
+TRTSERVER_Error*
+TRTSERVER_ServerGetSharedMemoryInfo(
```

Comment:
This needs a different name. It should just return an address.

Reply:
Renamed to GetSharedMemoryAddress and returning calculated `mapped_address + offset`